File: app/src/main/java/com/khmelenko/lab/travisclient/util/AssetsUtils.java
package com.khmelenko.lab.travisclient.util;

import android.content.Context;
import android.content.res.AssetManager;
import android.content.res.Resources;

import java.io.IOException;
import java.io.InputStream;
import java.util.Properties;

/**
 * Gives util methods for working with Assets
 *
 * @author Dmytro Khmelenko
 */
public final class AssetsUtils {

    // denied constructor
    private AssetsUtils() {
    }

    /**
     * Gets properties from Assets
     *
     * @param name    Properties file
     * @param context Context
     * @return Properties
     */
    public static Properties getProperties(String name, Context context) {
        Resources resources = context.getResources();
        AssetManager assetManager = resources.getAssets();
        Properties properties = null;

        // Read from the /assets directory
        try {
            InputStream inputStream = assetManager.open(name);
            properties = new Properties();
            properties.load(inputStream);
        } catch (IOException e) {
            e.printStackTrace();
        }
        return properties;
    }
}

File: app-v3/src/main/java/com/khmelenko/lab/varis/util/AssetsUtils.kt
package com.khmelenko.lab.varis.util

import android.content.Context
import java.io.IOException
import java.util.Properties

/**
 * Gives util methods for working with Assets
 *
 * @author Dmytro Khmelenko
 */
object AssetsUtils {

    /**
     * Gets properties from Assets
     *
     * @param name    Properties file
     * @param context Context
     * @return Properties
     */
    fun getProperties(name: String, context: Context): Properties? {
        val resources = context.resources
        val assetManager = resources.assets
        var properties: Properties? = null

        // Read from the /assets directory
        try {
            val inputStream = assetManager.open(name)
            properties = Properties()
            properties.load(inputStream)
        } catch (e: IOException) {
            e.printStackTrace()
        }

        return properties
    }
}
